core/src/main/java/aima/core/search/api/BidirectionalProblem.java

```diff
@@ -0,0 +1,20 @@
+package aima.core.search.api;
+
+/**
+ * @author manthan
+ */
+public interface BidirectionalProblem<A, S> extends
+        // S initialState()
+        InitialStateFunction<S>,
+        // List<A> actions(S s)
+        ActionsFunction<A, S>,
+        // List<S> results(S s, A a)
+        ResultFunction<A, S>,
+        // boolean isGoalState(S s)
+        GoalTestPredicate<S>,
+        // double stepCost(S s, A a, S s')
+        StepCostFunction<A, S> {
+    Problem<A,S> getOriginalProblem();
```

Comment:
BidirectionalProblem.initialState() and BidirectionalProblem.getOriginalProblem().initialState(), both will have same results. I made BidirectionalProblem.getOriginalProblem() for this purpose only. But I used it as a different problem to keep continuity with the superclasses of BasicProblem (which are all problems in themselves). 
Shall I replace BidirectionalProblem.getOriginalProblem().initialState() ? Your call on this aspect.